derive-new/src/lib.rs

```diff
@@ -52,7 +52,38 @@ fn new_for_struct(ast: syn::MacroInput) -> quote::Tokens {
                 }
             }
         },
-        syn::Body::Struct(syn::VariantData::Tuple(_)) => panic!("#[derive(new)] cannot be used with tuple structs"),
+        syn::Body::Struct(syn::VariantData::Tuple(ref fields)) => {
+            // Special-case a single member - call the parameter "value" instead of "value1"
+            if fields.len() == 1 {
+                let field = &fields[0];
+                let ty = &field.ty;
+
+                quote! {
+                    impl #impl_generics #name #ty_generics #where_clause {
+                        pub fn new(value: #ty) -> Self {
+                            #name(value)
+                        }
+                    }
+                }
+            } else {
+                let args = fields.iter().enumerate().map(|(i, f)| {
+                    let f_name = syn::Ident::new(format!("value{}", i + 1));
+                    let ty = &f.ty;
+                    quote!(#f_name: #ty)
+                });
+                let inits = fields.iter().enumerate().map(|(i, _)| {
+                    syn::Ident::new(format!("value{}", i + 1))
```

Comment:
I did try `let (args, inits) = (...).unzip();` but got an error that Rust couldn't infer the type of `args`. Let me try harder.